CRM/Report/BAO/ReportInstance.php

```diff
@@ -41,10 +41,16 @@ class CRM_Report_BAO_ReportInstance extends CRM_Report_DAO_ReportInstance {
    *
    * @param array $params
    *   (reference ) an assoc array of name/value pairs.
-   *
-   * @return CRM_Report_DAO_ReportInstance
+   * @return \CRM_Report_DAO_ReportInstance
+   * @throws \CRM_Core_Exception
    */
   public static function add(&$params) {
+    if (!CRM_Core_Permission::check('save Report Criteria')) {
```

Comment:
so to make the test pass I think this needs to be

if (CRM_Utils_Array::value('check_permissions') && !CRM_Core_Permission::check('save Report Criteria'))

Reply:
Alternatively this check could be stripped out & we could reply on the form - which is consistent with a lot of other places ( you indicate that it doesn't really work anyway)

CRM_Report_Info::getPermissions is the place to define api permissions